File: VB/WebSite/DevExpress/ASPxSpellCheckerForms/SpellCheckOptionsForm.ascx.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Collections
Imports System.Web
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls
Imports DevExpress.Web.ASPxSpellChecker
Imports DevExpress.Web.ASPxEditors
Imports DevExpress.Web.ASPxSpellChecker.Localization

Partial Public Class SpellCheckOptionsForm
	Inherits SpellCheckerUserControl
	Protected Overrides Sub PrepareChildControls()
		PrepareFormCheckBoxes()
		PrepareLanguagesCombo()
		Localize()

		MyBase.PrepareChildControls()
	End Sub

	Protected Overrides Function GetChildDxEdits() As ASPxEditBase()
		Return New ASPxEditBase() { chkbEmails, chkbMixedCase, chkbNumbers, chkbUpperCase, chkbUrls, comboLanguage, chkbTags }
	End Function

	Protected Overrides Function GetChildDxButtons() As ASPxButton()
		Return New ASPxButton() { btnCancel, btnOK }
	End Function

	Protected Sub PrepareFormCheckBoxes()
		Dim spellChecker As ASPxSpellChecker = (CType(NamingContainer, ASPxSpellChecker))
		chkbEmails.Checked = spellChecker.SettingsSpelling.IgnoreEmails
		chkbMixedCase.Checked = spellChecker.SettingsSpelling.IgnoreMixedCaseWords
		chkbNumbers.Checked = spellChecker.SettingsSpelling.IgnoreWordsWithNumber
		chkbUpperCase.Checked = spellChecker.SettingsSpelling.IgnoreUpperCaseWords
		chkbUrls.Checked = spellChecker.SettingsSpelling.IgnoreUrls
		chkbTags.Checked = spellChecker.SettingsSpelling.IgnoreMarkupTags
	End Sub
	Protected Sub PrepareLanguagesCombo()
		comboLanguage.Items.Clear()
		For i As Integer = 0 To SpellChecker.Dictionaries.Count - 1
			If comboLanguage.Items.FindByText(SpellChecker.Dictionaries(i).GetCulture().DisplayName) Is Nothing Then
				comboLanguage.Items.Add(SpellChecker.Dictionaries(i).GetCulture().DisplayName)
			End If
		Next i
		comboLanguage.SelectedItem = comboLanguage.Items.FindByText(SpellChecker.GetCulture().DisplayName)
	End Sub
	Protected Sub Localize()
		pnlOptions.HeaderText = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.GeneralOptions)
		chkbUpperCase.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.IgnoreUppercase)
		chkbMixedCase.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.IgnoreMixedCase)
		chkbNumbers.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.IgnoreNumbers)
		chkbEmails.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.IgnoreEmails)
		chkbUrls.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.IgnoreUrls)
		chkbTags.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.IgnoreTags)
		pnlLanguageSelection.HeaderText = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.InternationalDictionaries)
		lblChooseDictionary.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.ChooseDictionary)
		lblLanguage.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.Language)
		btnOK.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.OkButton)
		btnCancel.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.CancelButton)
	End Sub
End Class

File: CS/WebSite/DevExpress/ASPxSpellCheckerForms/SpellCheckOptionsForm.ascx.cs
using System;
using System.Data;
using System.Configuration;
using System.Collections;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
using DevExpress.Web.ASPxSpellChecker;
using DevExpress.Web.ASPxEditors;
using DevExpress.Web.ASPxSpellChecker.Localization;

public partial class SpellCheckOptionsForm : SpellCheckerUserControl {
    protected override void PrepareChildControls() {
        PrepareFormCheckBoxes();
        PrepareLanguagesCombo();
        Localize();

        base.PrepareChildControls();
    }

    protected override ASPxEditBase[] GetChildDxEdits() {
        return new ASPxEditBase[] { chkbEmails, 
                                    chkbMixedCase, 
                                    chkbNumbers, 
                                    chkbUpperCase, 
                                    chkbUrls, 
                                    comboLanguage, 
                                    chkbTags 
        };
    }

    protected override ASPxButton[] GetChildDxButtons() {
        return new ASPxButton[] {
            btnCancel,
            btnOK
        };
    }

    protected void PrepareFormCheckBoxes() {
        ASPxSpellChecker spellChecker = ((ASPxSpellChecker)NamingContainer);
        chkbEmails.Checked = spellChecker.SettingsSpelling.IgnoreEmails;
        chkbMixedCase.Checked = spellChecker.SettingsSpelling.IgnoreMixedCaseWords;
        chkbNumbers.Checked = spellChecker.SettingsSpelling.IgnoreWordsWithNumber;
        chkbUpperCase.Checked = spellChecker.SettingsSpelling.IgnoreUpperCaseWords;
        chkbUrls.Checked = spellChecker.SettingsSpelling.IgnoreUrls;
        chkbTags.Checked = spellChecker.SettingsSpelling.IgnoreMarkupTags;
    }
    protected void PrepareLanguagesCombo() {
        comboLanguage.Items.Clear();
        for (int i = 0; i < SpellChecker.Dictionaries.Count; i++) {
            if (comboLanguage.Items.FindByText(SpellChecker.Dictionaries[i].GetCulture().DisplayName) == null)
                comboLanguage.Items.Add(SpellChecker.Dictionaries[i].GetCulture().DisplayName);
        }
        comboLanguage.SelectedItem = comboLanguage.Items.FindByText(SpellChecker.GetCulture().DisplayName);
    }
	protected void Localize() {
		pnlOptions.HeaderText = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.GeneralOptions);
		chkbUpperCase.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.IgnoreUppercase);
        chkbMixedCase.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.IgnoreMixedCase);
        chkbNumbers.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.IgnoreNumbers);
		chkbEmails.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.IgnoreEmails);
        chkbUrls.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.IgnoreUrls);
		chkbTags.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.IgnoreTags);
		pnlLanguageSelection.HeaderText = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.InternationalDictionaries);
        lblChooseDictionary.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.ChooseDictionary);
        lblLanguage.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.Language);
		btnOK.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.OkButton);
		btnCancel.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.CancelButton);
	}
}
